### PR TITLE
merge of lean frontend into core

### PR DESCRIPTION
**Pull request by boutell**
Implements #1938. 

This is a WIP until I test what happens if you try to use the existing `apostrophe-lean-frontend` module with it. Probably nothing good, so I'll have to figure out a workaround to let that work before we merge.

But otherwise this works well, see the `lean-tests` branch of `apostrophe-enterprise-testbed`.

### CODE REVIEW

**Comment by boutell** [comment]
Testing shows this works acceptably with legacy apostrophe-lean-frontend present, it is now a candidate to be reviewed and merged.

Also fixed eslint warnings.

**Comment by boutell** [comment]
fixed dumb bug pls review

File: lib/modules/apostrophe-video-widgets/index.js [hunk 1]
```diff
@@ -24,6 +37,8 @@ module.exports = {
     ].concat(options.addFields || []);
   },
   construct: function(self, options) {
-    self.pushAsset('stylesheet', 'always', { when: 'always' });
+    if (self.apos.assets.options.lean && self.options.player) {
```
[the remark this answers is not in the document]
Comment (abea) [reply lib/modules/apostrophe-video-widgets/index.js]: @boutell One Q here.

[the remark this answers is not in the document]
Comment (boutell) [reply lib/modules/apostrophe-video-widgets/index.js]: D'oh. Thanks.

**Comment by abea** [comment]
Tom, I saw the testbed branch, but doesn't look like there are tests for it. Is that right?

**Comment by boutell** [comment]
Since I have been a persistent dummyface here I would appreciate review of this new commit which hopefully resolves everything you and I found. Thanks!

**Comment by boutell** [comment]
Correct, we do not have regression tests for it. This is an optional feature, apart from the library being present for code that wants it, which the existing regression tests will cover. We can add full regression tests of lean mode soon but I don't think it would be reasonable to hold things up another day just for that.

**Comment by boutell** [comment]
(Another day of developer time spent, that is.)

**Pull request merged**